File: sources/Backend/Models/Accounts/AccountModel.cs
namespace Lewio.CashFlow.Accounts;

public class AccountModel
{

   public EntityID AccountID { get; set; }

   public string Text { get; set; }

   public AccountTypeEnum Type { get; set; }

   public short? ClosingDay { get; set; }
   public short? DueDay { get; set; }
   public DateOnly? DueDate { get; set; }

   public bool Active { get; set; }

}

File: sources/Frontend/src/app/models/accounts/account.model.ts
import { AccountTypeEnum } from ".";

export class AccountModel {
   AccountID: string = '';

   Text: string = '';

   Type: AccountTypeEnum = AccountTypeEnum.General;

   ClosingDay?: number;
   DueDay?: number;
   DueDate?: Date;

   Active: boolean = false;

   public static Parse(value: AccountModel) {
      return Object.assign(new AccountModel, value);
   }
}
